routes/accounts/weekly_summary.js

```diff
@@ -0,0 +1,50 @@
+'use strict'
+const express = require( 'express' )
+const router = express.Router()
+const moment = require( 'moment' )
+const sequelize = require( 'sequelize' )
+
+
+const numberOfTasksCreated = ( Item, $gte, user_id ) =>
+  Item.findAndCountAll({ where: { createdAt: { $gte }, user_id } })
+    .then( result => result.count )
+
+
+const numberOfTasksCompleted = ( sequelize, user_id, since ) =>
+  sequelize.query(
+    `SELECT COUNT(DISTINCT item_id) FROM "Audits" WHERE field_name = 'completed' AND old_value = 'false' AND new_value = 'true' AND user_id = ${user_id} AND "createdAt" > '${since}'`,
+    { type: sequelize.QueryTypes.SELECT }
+  )
+
+const numberOfTasksEdited = ( sequelize, user_id, since ) =>
+  sequelize.query(
+    `SELECT COUNT(DISTINCT item_id) FROM "Audits" WHERE field_name = 'title' OR field_name ='description' AND old_value != new_value AND user_id = ${user_id} AND "createdAt" > '${since}'`,
+    { type: sequelize.QueryTypes.SELECT }
+  )
+
+const totalNumberOfTasks = ( Item, user_id ) =>
+  Item.findAndCountAll({ where: { user_id : user_id } })
+    .then( result => result.count )
+
+router.get( '/', (request, response ) => {
+  const { sequelize } = request.app.get( 'models' ).Item
+  const { Item, Audit } = request.app.get( 'models' )
+  const { user } = request
+
+  const displayLastMonday = moment().startOf('isoWeek').format('LL')
+  const lastMonday = moment().startOf('isoWeek').format('YYYY-MM-DD HH:mm:ss')
+
+
+  Promise.all([
+    numberOfTasksCreated( Item, lastMonday, user.id ),
+    numberOfTasksEdited( sequelize, user.id, lastMonday ),
+    numberOfTasksCompleted( sequelize, user.id, lastMonday ),
+    totalNumberOfTasks( Item, user.id )
+  ])
+  .then( ([ tasksCreated, tasksEdited, tasksCompleted, tasksTotal ]) =>
+    response.json({ tasksCreated, tasksEdited: tasksEdited[0].count, tasksCompleted: tasksCompleted[0].count, tasksTotal, displayLastMonday })
```

Comment:
We should return the appropriate data from the functions we defined above, instead of parsing it out at this point, and then use implicit object creation here (it's the responsibility of those functions to obtain these numbers, and we don't want the how of that to leak into further promises in the composition):
```js
  response.json({ tasksCreated, tasksEdited, tasksCompleted, tasksTotal })
```